### PR TITLE
Add underflow checking for TLVWriter::VPutStringF

### PR DESCRIPTION
**Pull request by holbrookt**

### CODE REVIEW

File: src/lib/core/WeaveTLVWriter.cpp [hunk 1]
```diff
@@ -878,7 +878,10 @@ WEAVE_ERROR TLVWriter::VPutStringF(uint64_t tag, const char *fmt, va_list ap)
     // no facilities for splitting the stream across multiple buffers,
     // just write however much fits in the current buffer.
     // assume conservative tag length at this time (8 bytes)
-    maxLen =  mRemainingLen - (1 + 8 + (1 << static_cast<uint8_t>(lenFieldSize)) + 1); // 1 : control byte, 8 : tag length, stringLen + 1 for null termination
+    size_t headerLen = (1 + 8 + (1 << static_cast<uint8_t>(lenFieldSize)) + 1); // 1 : control byte, 8 : tag length, stringLen + 1 for null termination
```
Comment (holbrookt) [inline src/lib/core/WeaveTLVWriter.cpp:881]: Is headerLen an accurate variable name here? Is that what this line is calculating?

File: src/lib/core/WeaveTLVWriter.cpp [hunk 2]
```diff
@@ -878,7 +878,10 @@ WEAVE_ERROR TLVWriter::VPutStringF(uint64_t tag, const char *fmt, va_list ap)
...
+    VerifyOrExit(mRemainingLen >= headerLen, err = WEAVE_ERROR_BUFFER_TOO_SMALL);
```
Comment (holbrookt) [inline src/lib/core/WeaveTLVWriter.cpp:882]: Is WEAVE_ERROR_BUFFER_TOO_SMALL an accurate error to return? If this check is false, it means that there is not enough space left

File: src/lib/core/WeaveTLVWriter.cpp (hunk 1 above)
Comment (robszewczyk) [reply src/lib/core/WeaveTLVWriter.cpp:881]: `elementHeadLen` would be more aligned with the terminology.

File: src/lib/core/WeaveTLVWriter.cpp (hunk 2 above)
Comment (robszewczyk) [reply src/lib/core/WeaveTLVWriter.cpp]: Yup, thats the right error

**Pull request merged**